MvvmCross/Core/Binding/Bindings/MvxFullBinding.cs

```diff
@@ -127,19 +128,23 @@ protected virtual void ClearTargetBinding()
         {
             if (this._targetBinding != null)
             {
-                if (this._targetBindingOnValueChanged != null)
+                lock (this._targetLocker)
                 {
-                    this._targetBinding.ValueChanged -= this._targetBindingOnValueChanged;
-                    this._targetBindingOnValueChanged = null;
-                }
+                    if (this._targetBindingOnValueChanged != null)
+                    {
+                        this._targetBinding.ValueChanged -= this._targetBindingOnValueChanged;
+                        this._targetBindingOnValueChanged = null;
+                    }
 
-                this._targetBinding.Dispose();
-                this._targetBinding = null;
+                    this._targetBinding.Dispose();
+                    this._targetBinding = null;
+                }
             }
         }
 
         private void CreateTargetBinding(object target)
         {
+            this._targetLocker = new object();
```

Comment:
Should we make this `readonly` and allocate in the constructor?  I'm worried that sometime in the future someone mind call `CreateTargetBinding` multiple times.